ci-operator/step-registry/interop-tests/cnv-tests-gs-baremetal/localnet/interop-tests-cnv-tests-gs-baremetal-localnet-commands.sh

```diff
@@ -0,0 +1,261 @@
+
+  # Delete these first since they might reference datavolumes or snapshots indirectly
+  oc delete datasources -n "${dvnamespace}" --selector='cdi.kubevirt.io/dataImportCron'
+
+  # Delete next because they might have dependencies on PVCs
+  oc delete datavolumes -n "${dvnamespace}" --selector='cdi.kubevirt.io/dataImportCron'
+
+  # Ugly hack for this external-snapshotter bug: https://github.com/kubernetes-csi/external-snapshotter/issues/1258.
+  local retry_count=0
+  local max_retries=10
+  local interval=30
+  while [[ $retry_count -lt $max_retries ]]; do
+      echo "Attempting to delete all volumesnapshots in namespace ${dvnamespace} (Attempt $((retry_count + 1)) of ${max_retries})..."
+
+      if oc delete volumesnapshots -n "${dvnamespace}" --selector=cdi.kubevirt.io/dataImportCron --timeout="${interval}s" --ignore-not-found; then
+          echo "Successfully deleted all volumesnapshots"
+          break
+      else
+          echo "Failed to delete some volumesnapshots. Trying to send dummy annotation to all dangling volumesnapshots"
+          retry_count=$((retry_count + 1))
+
+          # send dummy-annotation so the CSI-sidecar will send a DeleteSnapshot RPC
+          for name in $(oc get volumesnapshot -n "${dvnamespace}" --selector=cdi.kubevirt.io/dataImportCron -ojsonpath='{.items[*].metadata.name}'); do
+            # Unfortunately, VolumeSnapshotContent resources do not include the label selectors of their associated VolumeSnapshots
+            volumesnapshotcontent_name=$(oc get volumesnapshotcontent -o json | jq -r ".items[] | select(.spec.volumeSnapshotRef.name == \"$name\") | .metadata.name")
+            oc annotate volumesnapshotcontent "${volumesnapshotcontent_name}" example.com/dummy-annotation="This is a dummy annotation"
+          done
+      fi
+  done
+
+  if [[ $retry_count -ge $max_retries ]]; then
+    echo "failed to delete all volumesnapshot after $max_retries attempts."
+    exit 1
+  fi
+
+  # Finally, delete PVCs
+  oc delete pvc -n "${dvnamespace}" --selector='cdi.kubevirt.io/dataImportCron'
+
+  echo "[DEBUG] Enable DataImportCron"
+  cnv::toggle_common_boot_image_import "true"
+  sleep 10
+  echo "[DEBUG] Wait for DataImportCron to re-import volumes"
+  oc wait DataImportCron -n "${dvnamespace}" --all --for=condition=UpToDate --timeout=20m
+  echo "[DEBUG] Printing persistent volume claims"
+  oc get pvc -n "${dvnamespace}"
+}
+
+function install_yq_if_not_exists() {
+    # Install yq manually if not found in image
+    echo "Checking if yq exists"
+    cmd_yq="$(yq --version 2>/dev/null || true)"
+    if [ -n "$cmd_yq" ]; then
+        echo "yq version: $cmd_yq"
+    else
+        echo "Installing yq"
+        mkdir -p /tmp/bin
+        export PATH=$PATH:/tmp/bin/
+        curl -L "https://github.com/mikefarah/yq/releases/latest/download/yq_linux_$(uname -m | sed 's/aarch64/arm64/;s/x86_64/amd64/')" \
+         -o /tmp/bin/yq && chmod +x /tmp/bin/yq
+    fi
+}
+
```

Comment:
Let us install `yq` here if needed.
```suggestion
install_yq_if_not_exists

```